packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -0,0 +1,158 @@
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import {
+    CSSResultArray,
+    html,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import {
+    property,
+    state,
+} from '@spectrum-web-components/base/src/decorators.js';
+import { InputSegments } from '@spectrum-web-components/input-segments';
+
+import styles from './date-time-picker.css.js';
+
+import '@spectrum-web-components/calendar/sp-calendar.js';
+import '@spectrum-web-components/field-label/sp-field-label.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-calendar.js';
+import '@spectrum-web-components/overlay/sp-overlay.js';
+import '@spectrum-web-components/picker-button/sp-picker-button.js';
+import '@spectrum-web-components/popover/sp-popover.js';
+
+/**
+ * @element sp-date-time-picker
+ *
+ * @event change - Announces when a new date/time is defined by emitting a `Date` object
+ *
+ * @slot calendar-icon - The icon used in the calendar button
+ * @slot help-text - Default or non-negative help text to associate to your form element
+ * @slot negative-help-text - Negative help text to associate to your form element when `invalid`
+ */
+export class DateTimePicker extends InputSegments {
+    public static override get styles(): CSSResultArray {
+        return [...super.styles, styles];
+    }
+
+    /**
+     * Indicates whether the picker should be displayed or not
+     */
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @state()
+    override includeDate = true;
+
+    @state()
+    override includeTime = true;
+
+    @state()
+    private pickerDate?: Date;
+
+    protected override renderField(): TemplateResult {
+        return html`
+            ${super.renderInputContent()} ${this.renderPicker()}
+        `;
+    }
+
+    public renderPicker(): TemplateResult {
+        const isDisabled = super.disabled || super.readonly;
+
+        return html`
+            <div class="picker">
+                ${super.renderStateIcons()}
+
+                <sp-picker-button
+                    ?open=${this.open}
+                    ?quiet=${super.quiet}
+                    ?invalid=${super.invalid}
+                    ?disabled=${isDisabled}
+                    @click=${this.openPicker}
+                >
+                    <slot name="calendar-icon" slot="icon">
+                        <sp-icon-calendar></sp-icon-calendar>
+                    </slot>
+                </sp-picker-button>
+
+                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
+                <sp-overlay
+                    placement="bottom-start"
```

Comment:
As per https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/ we likely want this to trap the tab order... _sometimes_. Can you add reviewing this to the to do list?

Generally, this is added via `type="modal"` but this _always_ traps the tab order, and we only want to do so _if_ the user enters the picker menu, not when the picker menu opens, which would be the current functionality.

Reply:
Added